core/arch/aarch64/codec.txt

```diff
@@ -970,4 +970,110 @@ x101101011000000000101xxxxxxxxxx  cls     wx0 : wx5
 1001111001100111000000xxxxxxxxxx     fmov d0 : x5
 1001111010101111000000xxxxxxxxxx     fmov q0 : x5 # only sets the bit top half of q0
 
+# Advanced SIMD three same (FP16)
+0x001110010xxxxx000001xxxxxxxxxx     fmaxnm    dq0 : dq5 dq16 h_sz
+0x001110010xxxxx000011xxxxxxxxxx     fmla      dq0 : dq0 dq5 dq16 h_sz
+0x001110010xxxxx000101xxxxxxxxxx     fadd      dq0 : dq5 dq16 h_sz
+0x001110010xxxxx000111xxxxxxxxxx     fmulx     dq0 : dq5 dq16 h_sz
+0x001110010xxxxx001001xxxxxxxxxx     fcmeq     dq0 : dq5 dq16 h_sz
+0x001110010xxxxx001101xxxxxxxxxx     fmax      dq0 : dq5 dq16 h_sz
+0x001110010xxxxx001111xxxxxxxxxx     frecps    dq0 : dq5 dq16 h_sz
+0x001110110xxxxx000001xxxxxxxxxx     fminnm    dq0 : dq5 dq16 h_sz
+0x001110110xxxxx000011xxxxxxxxxx     fmls      dq0 : dq0 dq5 dq16 h_sz
+0x001110110xxxxx000101xxxxxxxxxx     fsub      dq0 : dq5 dq16 h_sz
+0x001110110xxxxx001101xxxxxxxxxx     fmin      dq0 : dq5 dq16 h_sz
+0x001110110xxxxx001111xxxxxxxxxx     frsqrts   dq0 : dq5 dq16 h_sz
```

Comment:
I picked this one randomly.  I'm looking in ARMv8 manual DDI 0487A.a section C6.3.145 "FRSQRTS" page C6-1006 and it lists the encoding top bits for scalar as 0101 and vector as 0Q001 -- neither of which looks like 00111.  Am I missing something?

Reply:
The top 8 bits for frsqrts vector (single + double) and vector (half) are  `0Q001110`. The top 8 bits in the pattern are  `0x001110`. In the format, an `x` stands for a bit that's set depending on the operands. In that case, it is the Q bit, which is set by the dqXX operands. Does that make sense?

Reply:
Oh, wow, my brain saw a leading "0x" and skipped right over it: interpreted it as a hex format marker.  Even now I have to force myself to include "0x" in the bits that matter.

Reply:
Yep I suppose it is slightly misleading in this context :) Thank you  very much for taking a careful look at this!